### PR TITLE
[MobileBert P0] Add PretrainedConfig and unit tests

### PR DESCRIPTION
**Pull request by LemonNoel**

### CODE REVIEW

File: paddlenlp/transformers/mobilebert/modeling.py [hunk 1]
```diff
@@ -52,34 +68,26 @@ class MobileBertEmbeddings(nn.Layer):
...
         self,
-        vocab_size,
-        embedding_size=128,
-        hidden_size=512,
-        hidden_dropout_prob=0.0,
-        max_position_embeddings=512,
-        type_vocab_size=2,
-        layer_norm_eps=1e-12,
+        config,
         pad_token_id=1,
```
Comment (sijunhe) [inline paddlenlp/transformers/mobilebert/modeling.py]: config里也有pad_token_id

File: paddlenlp/transformers/mobilebert/modeling.py [hunk 2]
```diff
@@ -779,7 +545,7 @@ def _init_weights(self, layer):
                     mean=0.0,
                     std=self.initializer_range
                     if hasattr(self, "initializer_range")
-                    else self.mobilebert.config["initializer_range"],
+                    else self.mobilebert.config.initializer_range,
```
Comment (sijunhe) [inline paddlenlp/transformers/mobilebert/modeling.py]: 这里可以不用if else了，直接`self.config.initializer_range`

File: paddlenlp/transformers/mobilebert/modeling.py [hunk 3]
```diff
@@ -1297,11 +1078,11 @@ class MobileBertForQuestionAnswering(MobileBertPretrainedModel):
...
 
-    def __init__(self, mobilebert):
-        super(MobileBertForQuestionAnswering, self).__init__()
+    def __init__(self, config):
+        super(MobileBertForQuestionAnswering, self).__init__(config)
         self.num_labels = 2
-        self.mobilebert = mobilebert
-        self.qa_outputs = nn.Linear(self.mobilebert.config["hidden_size"], self.num_labels)
+        self.mobilebert = MobileBertModel(config)
+        self.qa_outputs = nn.Linear(self.mobilebert.config.hidden_size, self.num_labels)
```
Comment (sijunhe) [inline paddlenlp/transformers/mobilebert/modeling.py]: self.config.hidden_size

File: paddlenlp/transformers/mobilebert/modeling.py [hunk 4]
```diff
@@ -799,16 +596,10 @@ class MobileBertForPreTraining(MobileBertPretrainedModel):
...
-            self.mobilebert.config["vocab_size"],
-            self.mobilebert.config["hidden_size"],
-            self.mobilebert.config["embedding_size"],
-            self.mobilebert.config["hidden_act"],
-            self.mobilebert.config["layer_norm_eps"],
-        )
+    def __init__(self, config):
+        super(MobileBertForPreTraining, self).__init__(config)
+        self.mobilebert = MobileBertModel(config)
+        self.cls = MobileBertPreTrainingHeads(self.mobilebert.config)
```
Comment (sijunhe) [inline paddlenlp/transformers/mobilebert/modeling.py]:
```suggestion
        self.cls = MobileBertPreTrainingHeads(config)
```

File: tests/transformers/mobile_bert/test_modeling.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+            input_ids,
+            token_type_ids,
+            input_mask,
+            sequence_labels,
+        ) = config_and_inputs
+        inputs_dict = {"input_ids": input_ids, "token_type_ids": token_type_ids, "attention_mask": input_mask}
+        return config, inputs_dict
+
+
+class MobileBertModelTest(ModelTesterMixin, unittest.TestCase):
```
Comment (sijunhe) [inline tests/transformers/mobile_bert/test_modeling.py:193]: 既然已经实现了return_dict, 单测里面可以仿照https://github.com/PaddlePaddle/PaddleNLP/blob/develop/tests/transformers/ernie/test_modeling.py#L372 测试return_dict正确性

File: tests/transformers/mobile_bert/test_modeling.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
+
+    def test_model_from_pretrained(self):
+        for model_name in MobileBertModel.pretrained_init_configuration.keys():
+            model = MobileBertModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+class MobileBertCompatibilityTest(unittest.TestCase):
```
Comment (sijunhe) [inline tests/transformers/mobile_bert/test_modeling.py]: 因为没有写AutoConverter, `MobileBertCompatibilityTest`不需要。如果需要测试已有的模型，可以模仿
https://github.com/PaddlePaddle/PaddleNLP/blob/develop/tests/transformers/ernie/test_modeling.py#L440 写一个MobileBertIntegrationTest即可

**Review by sijunhe — COMMENTED** [review]
solid PR. a few minor changes and we are good to go!

File: paddlenlp/transformers/mobilebert/modeling.py (hunk 1 above)
Comment (LemonNoel) [reply paddlenlp/transformers/mobilebert/modeling.py]: 改为 config.pad_token_id.

File: paddlenlp/transformers/mobilebert/modeling.py (hunk 2 above)
Comment (LemonNoel) [reply paddlenlp/transformers/mobilebert/modeling.py]: 已改为 `self.config.initializer_range`

File: paddlenlp/transformers/mobilebert/modeling.py (hunk 3 above)
Comment (LemonNoel) [reply paddlenlp/transformers/mobilebert/modeling.py]: Done

File: paddlenlp/transformers/mobilebert/modeling.py (hunk 4 above)
Comment (LemonNoel) [reply paddlenlp/transformers/mobilebert/modeling.py]: Done

File: tests/transformers/mobile_bert/test_modeling.py (hunk 5 above)
Comment (LemonNoel) [reply tests/transformers/mobile_bert/test_modeling.py:193]: 已增加 `return_dict` 测试

File: tests/transformers/mobile_bert/test_modeling.py (hunk 6 above)
Comment (LemonNoel) [reply tests/transformers/mobile_bert/test_modeling.py]: 删掉了 `MobileBertCompatibilityTest`，增加了 `MobileBertIntegrationTest`

**Review by sijunhe — APPROVED** [review]
lgtm. great job!

**Pull request merged**